src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -390,20 +390,22 @@ impl Net {
     /// https://docs.oasis-open.org/virtio/virtio/v1.1/csprd01/virtio-v1.1-csprd01.html#x1-320005
     /// 2.6.7.1 Driver Requirements: Used Buffer Notification Suppression
     fn try_signal_queue(&mut self, queue_type: NetQueue) -> Result<(), DeviceError> {
-        let queue = match queue_type {
-            NetQueue::Rx => &mut self.queues[RX_INDEX],
-            NetQueue::Tx => &mut self.queues[TX_INDEX],
+        let qidx = match queue_type {
+            NetQueue::Rx => RX_INDEX,
+            NetQueue::Tx => TX_INDEX,
         };
 
-        if queue.prepare_kick() {
+        if self.queues[qidx].prepare_kick() {
             let (_, interrupt) = self
                 .device_state
                 .active_state()
                 .expect("Device is not initialized");
-            interrupt.trigger_irq(IrqType::Vring).map_err(|err| {
-                self.metrics.event_fails.inc();
-                DeviceError::FailedSignalingIrq(err)
-            })?;
+            interrupt
+                .trigger(VirtioInterruptType::Queue(qidx.try_into().unwrap()))
```

Comment:
We really need to make queue indexes `u16`. Otherwise it is painful to watch comptime constants needing this treatment.

Reply:
Yup, agreed. But I don't think that's part of this PR